src/rgw/rgw_http_client_ssl.h

```diff
@@ -0,0 +1,54 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2018 SUSE Linux GmBH
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef RGW_HTTP_CLIENT_SSL_H
+#define RGW_HTTP_CLIENT_SSL_H
+
+#include <mutex>
+#include <vector>
+#include "common/ceph_context.h"
+#include "common/dout.h"
+#include "rgw_common.h"
+
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_rgw
+
+void rgw_ssl_locking_callback(int mode, int n, const char *file, int line);
+unsigned long rgw_ssl_thread_id_callback();
+
+
+class RGWSSLSetup
```

Comment:
this class, along with the dout defines/includes, can live in the source file instead

Reply:
ack